e2e/playwright/fixtures/sceneFixture.ts

```diff
@@ -89,4 +90,28 @@ export class SceneFixture {
   waitForExecutionDone = async () => {
     await expect(this.exeIndicator).toBeVisible()
   }
+
+  expectPixelColor = async (
+    colour: [number, number, number],
+    coords: { x: number; y: number },
+    diff: number
+  ) => {
+    let finalValue = colour
+    await expect
+      .poll(async () => {
+        const pixel = (await getPixelRGBs(this.page)(coords, 1))[0]
+        if (!pixel) return null
+        finalValue = pixel
+        return pixel.every(
+          (channel, index) => Math.abs(channel - colour[index]) < diff
+        )
+      })
+      .toBeTruthy()
+      .catch((e) => {
+        console.error(
+          `Error is ExpectPixelColor: expecting ${colour} got ${finalValue}`
+        )
+        throw e
```

Comment:
"Log and throw" is generally known as an anti-pattern because if you're throwing, presumably something else is going to handle that. But if you log also, then you've duplicated the error.

You should be able to do something like this based on the [MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#differentiate_between_similar_errors):

```typescript
throw new Error(`Error is ExpectPixelColor: expecting ${colour} got ${finalValue}`, { cause: e })
```

But up to you. It's just a test.

Reply:
I was half conscious of that yeah, might as well be consistent, I'm going to have to make another commit to fix the windows thing, so I,ll fix this up too.